controllers/front/CartController.php

```diff
@@ -617,14 +606,8 @@ protected function shouldAvailabilityErrorBeRaised($product, $qtyToCheck)
 
         /*
          * We check if this product is out-of-stock.
-         *
-         * @todo StockAvailable::getQuantityAvailableByProduct does not work for packs
-         * which depend on quantity of products inside.
          */
-        $availableProductQuantity = StockAvailable::getQuantityAvailableByProduct(
-            $this->id_product,
-            $this->id_product_attribute
-        );
+        $availableProductQuantity = Product::getQuantity($this->id_product, $this->id_product_attribute);
```

Comment:
This is actually the real fix. Product::getQuantity is just a "passthrough" to `StockAvailable::getQuantityAvailableByProduct` for normal products, but will provide a proper pack quantity. 👍

The change is safe, we are not passing a cart or customization.